### DISCUSSION TITLE
Video Recording Software (GIFs)

### DISCUSSION THREAD

**Discussion by gs-101** [minad/tempel#148] [Q&A] [answered]
What software do you use in recording [these](https://github.com/minad/tempel/tree/screenshots) GIFs? I ~try to~ use OBS for recording but every time I upload it somewhere compression just ruins the video.
I actually just tested uploading in GitHub, and that doesn't seem to be an issue here, so if you just use OBS then I can probably just convert the video with FFMPEG.

**Comment by minad** [comment, marked as answer]
Iirc I made individual screenshots of each frame, via some Elisp helper script.

**Reply by gs-101** [reply]
Thanks, I don't know how I'd do it this way but I'll try some different alternatives, and list them here for the sake of discussion and sharing alternatives for recording:

- [x] [gifski](https://gif.ski/) [software]
- [ ] [gif-screencast](https://gitlab.com/ambrevar/emacs-gif-screencast) [package]
   - Got an error with ImageMagick
- [x] [ffmpeg conversion](https://www.baeldung.com/linux/convert-videos-gifs-ffmpeg) [conversion]
  - Slower and blurrier than gifski, but doesn't require stalling additional software
  - Command: `input.mp4 -i palette.png -filter_complex "[0:v] fps=60,scale=1080:-1 [new];[new][1:v] paletteuse" output.gif`
- [x] [vhs](https://github.com/charmbracelet/vhs) [software, tty only]

The ones with boxes checked are options I approve. The best one for now was gifski.
